modules/standard/Set.chpl

```diff
@@ -306,14 +313,19 @@ module Set {
       var result = false;
 
       on this {
-        _enter();
+        _enter(); defer _leave();
 
-        if _dom.contains(x) {
-          _dom.remove(x);
+        var (hasFoundSlot, idx) = _htb.findFullSlot(x);
+
+        if hasFoundSlot {
+          pragma "no init"
```

Comment:
You should not need this pragma; split-init with `out` intent should take care of it. (A good idea to write tests to check though).

Reply:
Ok, thanks!